Web Quiz Engine (Kotlin)/task/src/engine/service/QuizService.kt

```diff
@@ -6,34 +6,62 @@ import engine.model.QuizPostResponseBodyCorrect
 import engine.model.QuizPostResponseBodyIncorrect
 import engine.exception.QuizNotFoundException
 import engine.entity.Quiz
+import engine.model.QuizDto
 import org.springframework.stereotype.Service
 
 @Service
-class QuizService(private val quizRepository: WebQuizRepository) {
+class QuizService(
+    private val quizRepository: WebQuizRepository,
+) {
 
-    fun findQuizById(
-        id: Int,
-        omitId: Boolean = false,
-        omitAnswerList: Boolean = false
-        ) =
-        quizRepository.getQuizById(
-            id = id,
-            omitId = omitId,
-            omitAnswerList = omitAnswerList,
-    )?: throw QuizNotFoundException(quizId = id)
+    fun findQuizById(id: Int): QuizDto =
+        quizRepository
+            .findById(id)
+            .orElseThrow { QuizNotFoundException(quizId = id) }
+            .let {
+                with(it) {
+                    QuizDto(
+                        id = id,
+                        title = title,
+                        text = text,
+                        optionList = optionList,
+                    )
+                }
+            }
 
-    fun findQuizList(omitAnswerList: Boolean = false) =
-        quizRepository.getAllQuizList(omitAnswerList = omitAnswerList)
+    fun findQuizList(): List<QuizDto> =
+        quizRepository.findAll().map { quiz ->
+            with(quiz) {
+                QuizDto(
+                    id = id,
+                    title = title,
+                    text = text,
+                    optionList = optionList,
+                )
+            }
+        }
 
     fun checkQuizSolution(quizId: Int, answerList: List<Int>?): QuizPostResponseBody =
-        quizRepository.getQuizById(id = quizId)?.let { quiz ->
+        quizRepository.findById(quizId)
```

Comment:
Lines 45-46 are exactly the same as lines 18-19-20.
You should write a common method to extract this logic and re-use it in those two places.